pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -637,6 +637,20 @@ func (k *Kaniko) resolveFailFast(ctx context.Context,
 				continue
 			}
 			if jobPod.Status.Phase == v1.PodPending || jobPod.Status.Phase == v1.PodUnknown {
+				if failure, failed := k.getLastPodFailureEvent(ctx, namespace, jobPod.Name); failed {
+					errorMessage := fmt.Sprintf("%s event for Kaniko pod %s. Message: %s",
+						failure.Reason,
+						jobPod.Name,
+						failure.Message)
+
+					// if an error has changed, print it to the logs
+					if errorMessage != lastError {
+						buildLogger.WarnWithCtx(ctx, fmt.Sprintf("%s event for Kaniko pod", failure.Reason),
+							"podName", jobPod.Name,
+							"message", failure.Message)
```

Comment:
This is different than your screenshot in the description.
Either way, I think it's better to day "Kaniko pod is not yet scheduled", then put the reason and message in the var args.

Reply:
@TomerShor it seems that we can't use `message` param here, because it replaces the actual message:
```
{"level":"warn","time":"2024-08-21T08:01:20.563Z","name":"dashboard.server.api/functions","message":"FailedScheduling event for Kaniko pod","requestID":"c1cf61d7a82c2691fbc1c51b57d3f562","more":{"message":"0/1 nodes are available: 1 node(s) didn't match Pod's node affinity/selector. preemption: not eligible due to preemptionPolicy=Never..","podName":"nuclio-kanikojob.nucliodefaulttestprocessorlatest.yfc094uip8xcm"}}
```

Reply:
@TomerShor it can be not only a scheduling issue, so we can't just say here `Kaniko pod is not yet scheduled`

Reply:
I renamed message to reason and updated the PR description with new version